### PR TITLE
[5.5] Make Request::validate() return value of parent key

### PR DESCRIPTION
**Pull request by themsaid**
In reference to the issue in https://github.com/laravel/framework/issues/20973

This PR will make the `Request::validate()` method return values of the parent key only, so if there's a rule `users.*.name`, the method returns the values of the `users` key only.

### CODE REVIEW

File: src/Illuminate/Foundation/Providers/FoundationServiceProvider.php [hunk 1]
```diff
@@ -38,7 +39,10 @@ public function registerRequestValidate()
         Request::macro('validate', function (array $rules, ...$params) {
             validator()->validate($this->all(), $rules, ...$params);
 
-            return $this->only(array_keys($rules));
+
+            return $this->only(collect($rules)->keys()->map(function($rule){
+                return Str::contains($rule, '.') ? explode('.', $rule)[0] : $rule;
```
Comment (JosephSilber) [inline src/Illuminate/Foundation/Providers/FoundationServiceProvider.php]: We have nice helpers these days:

```php
return str_before($rule, '.');
```

**Comment by JosephSilber** [comment]
This also has to be done in ValidatesRequests: https://github.com/laravel/framework/blob/5.5/src/Illuminate/Foundation/Validation/ValidatesRequests.php#L29-L49

As well as in form requests: https://github.com/laravel/framework/blob/5.5/src/Illuminate/Foundation/Http/FormRequest.php#L173-L178

**Comment by themsaid** [comment]
@JosephSilber handled

**Comment by MCMatters** [comment]
I am surprised. It is exactly what I told you previously. How do you decide to make these "features"? You did not agree with me and now I see "another picture"

**Comment by taylorotwell** [comment]
@MCMatters sometimes people have to review and think about things.

**Pull request merged**